inspire/testsuite/test_latex.py

```diff
@@ -0,0 +1,127 @@
+#
+# INSPIRE is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+import pkg_resources
+import os
+
+from dojson.contrib.marc21.utils import create_record
+
+from invenio.testsuite import InvenioTestCase, make_test_suite, run_test_suite
+from inspire.dojson.hep import hep
+from invenio.base.wrappers import lazy_import
+Latex = lazy_import('inspire.utils.latex.Latex')
+
+
+class LatexTests(InvenioTestCase):
+
+    def setUp(self):
+        self.marcxml = pkg_resources.resource_string('inspire.testsuite',
+                                                     os.path.join(
+                                                         'fixtures',
+                                                         'test_hep_formats.xml')
+                                                     )
+        record = create_record(self.marcxml)
+
+        self.hep_record = hep.do(record)
+
+        self.sample_latex_eu = {
+            'citation_key': 'Aad:2015wqa',
+            'author': 'G.~Aad',
+            'title': '\nSearch for supersymmetry in events containing a same-flavour opposite-sign dilepton pair, jets, and large missing transverse momentum in $\sqrt{s}=8$ TeV $pp$ collisions with the ATLAS detector\n',
+            'publi_info': ['Eur.\ Phys.\ J.\ C {\\bf 75} (2015) 7,  318', '[Eur.\ Phys.\ J.\ C {\\bf 75} (2015) 10,  463]'],
+            'arxiv': 'arXiv:1503.03290 [hep-ex]',
+            'report_number': '',
+            'SLACcitation': '%%CITATION = ARXIV:1503.03290;%%',
+
+        }
+
+        self.sample_latex_us = {
+            'citation_key': 'Aad:2015wqa',
+            'author': 'G.~Aad',
+            'title': '\nSearch for supersymmetry in events containing a same-flavour opposite-sign dilepton pair, jets, and large missing transverse momentum in $\sqrt{s}=8$ TeV $pp$ collisions with the ATLAS detector\n',
+            'publi_info': ['Eur.\ Phys.\ J.\ C {\\bf 75}, no. 7, 318 (2015)', '[Eur.\ Phys.\ J.\ C {\\bf 75}, no. 10, 463 (2015)]'],
+            'arxiv': 'arXiv:1503.03290 [hep-ex]',
+            'report_number': '',
+            'SLACcitation': '%%CITATION = ARXIV:1503.03290;%%',
+
+        }
+
+    def test_citation_key(self):
+        """Test if citation_key is created correctly"""
+        self.assertEqual(self.sample_latex_eu['citation_key'],
+                         Latex(self.hep_record, 'latex_eu').
```

Comment:
Same as before, you can do in the `setUp`

```python
Latex(self.hep_record, 'latex_eu')
Latex(self.hep_record, 'latex_us')
```

only once and then use it in the tests.